### PR TITLE
Split oc and gitserver into their own binaries

### PR DESCRIPTION
**Pull request by smarterclayton**
This begins to provide a separate oc binary which can be smaller
than the origin binary.

[test]

@fabianofranz review

### CODE REVIEW

File: cmd/gitserver/gitserver.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+func main() {
+	defer serviceability.BehaviorOnPanic(os.Getenv("OPENSHIFT_ON_PANIC"))()
+	defer serviceability.Profile(os.Getenv("OPENSHIFT_PROFILE")).Stop()
+
+	if len(os.Getenv("GOMAXPROCS")) == 0 {
+		runtime.GOMAXPROCS(runtime.NumCPU())
+	}
+
+	basename := filepath.Base(os.Args[0])
+	command := gitserver.NewCommandGitServer(basename)
```
Comment (fabianofranz) [inline cmd/gitserver/gitserver.go]: `cli.CommandFor` here and `gitserver.NewCommandGitServer` in the body of `CommandFor`?

Comment (fabianofranz) [inline cmd/gitserver/gitserver.go]: And in that case the complete body of `main` could be reused.

**Comment by fabianofranz** [comment]
I see although compiling `oc` you still want to support `oc` and `osc` symlinks to `openshift`, correct? In that case 2 comments then LGTM. 
I didn't think too much but this *may* result in some doc change(s).

**Comment by smarterclayton** [comment]
For now

On Mon, Jul 13, 2015 at 2:22 PM, Fabiano Franz <notifications@github.com>
wrote:

> I see although compiling oc you still want to support oc and osc symlinks
> to openshift, correct? In that case 2 comments then LGTM.
> I didn't think too much but this *may* result in some doc change(s).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/3685#issuecomment-121013907>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift

File: cmd/gitserver/gitserver.go (hunk 1 above)
Comment (smarterclayton) [inline cmd/gitserver/gitserver.go]: gitserver shouldn't import CLI

Comment (fabianofranz) [inline cmd/gitserver/gitserver.go]: Right.

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**